sdk/identity/azure-identity/test/ut/azure_cli_credential_test.cpp

```diff
@@ -570,6 +575,40 @@ TEST(AzureCliCredential, StrictIso8601TimeFormat)
       DateTime::Parse("2022-08-24T00:43:08.000000Z", DateTime::DateFormat::Rfc3339));
 }
 
+TEST(AzureCliCredential, LocalTime)
+{
+  constexpr auto Token = "{\"accessToken\":\"ABCDEFGHIJKLMNOPQRSTUVWXYZ\","
+                         "\"expiresOn\":\"2023-12-07 00:43:08\"}";
```

Comment:
What happens if the input expiresOn within the JSON has trailing space (or some other semantic text that is typically ignored), and looks like this:
`"\"expiresOn\":\"2023-12-07 00:43:08   \"}"`

I imagine, this would previously work OK, but now the DateTime parser won't handle it, once the local time is appended after the spaces.

Reply:
If they start giving bad outputs, we can build some vandal-proofness, but I believe failing is not the worst option.
I do not expect them to start returning values with spaces at the end, though.

Reply:
BTW, if we ever need to do this, one way of doing it would be to parse it, and then do a ToString(), then remove `Z` at the end, then append a TZ offset string, and then parse again.
Should there be a TZ offset, it would be harder to detect it, without writing a parser, so I think if that is ever the case, the best course of action would be the initial state of this PR, when it was the DateTime's parser who would append TZ offset IF there is no offset or `Z` specified, and would ignore it otherwise.